Update for Netbeans 25

aliedperezmartinez opened the pull request:
It seems to be OK. I would update the simple project template.

aliedperezmartinez committed:
update to NB 25

- Set NB release 25
- Update all dependencies
- Migrate tests to JUnit 5
- Add Yenta class into code

aliedperezmartinez committed:
Set version to 4.0

aliedperezmartinez committed:
Update main.yml

aliedperezmartinez committed:
Update main.yml with distribution

aliedperezmartinez committed:
Update main.yml

aliedperezmartinez committed:
publish packages

aliedperezmartinez committed:
Update pom.xml

aliedperezmartinez committed:
Merge branch 'master' of https://github.com/aliedperezmartinez/nb-springboot

aliedperezmartinez commented:
- Set NB release 25
- Update all dependencies
- Migrate tests to `JUnit` 5
- Add `Yenta` class into code (couldn't find it in a repository)

aliedperezmartinez committed:
Update pom.xml

aliedperezmartinez commented on pom.xml:
```diff
@@ -275,23 +276,29 @@
             <version>${netbeans.api.version}</version>
         </dependency>
-        <dependency>
+        <!--        <dependency>
```
Copied the file `Yenta.java` to this project. I couldn't find it in a repository.

aliedperezmartinez committed:
remove spring classes and include it as dependency

aliedperezmartinez committed:
Refactors and code updates

aliedperezmartinez committed:
more refactor

aliedperezmartinez committed:
Use NamedItem record accessors

aliedperezmartinez committed:
small refactors

AlexFalappa commented:
@aliedperezmartinez thanks for the PR, can you summarize what have you done here and why?

Is this purely an update of the codebase to a newer Java language version and to updated dependencies or is this needed to make the plugin work with the latest NetBeans version?

I regret to inform you that I have stopped maintaining the plugin a while ago.

aliedperezmartinez commented:
> @aliedperezmartinez thanks for the PR, can you summarize what have you done here and why?
> 
> Is this purely an update of the codebase to a newer Java language version and to updated dependencies or is this needed to make the plugin work with the latest NetBeans version?
> 
> I regret to inform you that I have stopped maintaining the plugin a while ago.

Hi @AlexFalappa ,
yes, this is not just the Java version and updated dependencies, but also to make it work with the latest (and greatest) NetBeans version. I also added a few refactors, and removed some classes from SpringBoot, after adding the dependency.

aliedperezmartinez commented:
I also want to add support for Gradle at some point.

aliedperezmartinez committed:
update jacoco

Exclude generated sources

Uthanien commented:
@aliedperezmartinez Sorry for asking, but maybe you can maintain a separate branch yourself? This plugin is quite useful for those of us who continue to use NetBeans.
greetings!

aliedperezmartinez commented:
> @aliedperezmartinez Sorry for asking, but maybe you can maintain a separate branch yourself? This plugin is quite useful for those of us who continue to use NetBeans. greetings!

I've created a fork and I can maintain it. I wanted first to ask the developers.

Uthanien commented:
> > @aliedperezmartinez Sorry for asking, but maybe you can maintain a separate branch yourself? This plugin is quite useful for those of us who continue to use NetBeans. greetings!
> 
> I've created a fork and I can maintain it. I wanted first to ask the developers.

I started following you, if you create a new repo, I can help you with some things, updating files and other things.
Greetings!

aliedperezmartinez commented:
> > > @aliedperezmartinez Sorry for asking, but maybe you can maintain a separate branch yourself? This plugin is quite useful for those of us who continue to use NetBeans. greetings!
> > 
> > 
> > I've created a fork and I can maintain it. I wanted first to ask the developers.
> 
> I started following you, if you create a new repo, I can help you with some things, updating files and other things. Greetings!

dale, weón :+1: :stuck_out_tongue:

Uthanien commented:
jajaja me avisas si quieres para seguir el nuevo repo para ir haciendo
algunas cosas ;)

El mié, 7 may 2025 a las 23:10, Alied Perez Martinez (<
***@***.***>) escribió:

> *aliedperezmartinez* left a comment (AlexFalappa/nb-springboot#185)
> <https://github.com/AlexFalappa/nb-springboot/pull/185#issuecomment-2861504021>
>
> @aliedperezmartinez <https://github.com/aliedperezmartinez> Sorry for
> asking, but maybe you can maintain a separate branch yourself? This plugin
> is quite useful for those of us who continue to use NetBeans. greetings!
>
> I've created a fork and I can maintain it. I wanted first to ask the
> developers.
>
> I started following you, if you create a new repo, I can help you with
> some things, updating files and other things. Greetings!
>
> dale, weón 👍 😛
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/AlexFalappa/nb-springboot/pull/185#issuecomment-2861504021>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABL7TW6EHOUX7M3YPPWQNLT25LDLPAVCNFSM6AAAAABZOYOBSCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDQNRRGUYDIMBSGE>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
*Felipe Espinosa Cancino*
Ingeniero en Informática

aliedperezmartinez commented:
OK, I'll close this PR, will continue in my fork[1]. Let me know if there's anything else I can do to help this project.
Thanks @AlexFalappa , for starting this.

[1] https://github.com/aliedperezmartinez/nb-springboot
